stock_dynamic_routing/models/stock_move.py

```diff
@@ -287,13 +287,19 @@ def _apply_routing_rule_pull(self, routing_details):
             # pull routing rules
             original_destination = move.location_dest_id
             current_picking_type = move.picking_id.picking_type_id
-            move.with_context(
-                __applying_routing_rule=True
-            ).location_id = routing_rule.location_src_id
-            move.picking_type_id = routing_rule.picking_type_id
-            dest_location = move.location_dest_id
-            rule_location = routing_rule.location_dest_id
-            if rule_location.is_sublocation_of(dest_location):
+
+            # Use the source location of the routing rule, if not already done
+            # If a sublocation is used, it's respected.
+            if not move.location_id._child_of(routing_rule.location_src_id):
+                move.with_context(
+                    __applying_routing_rule=True
+                ).location_id = routing_rule.location_src_id
```

Comment:
Can you complete a test with this fix? Validate the pulled move and check the chain is not broken

Reply:
Sure thing, it's done!